crates/sui-json-rpc/src/api.rs

```diff
@@ -97,6 +97,18 @@ pub trait RpcReadApi {
         object_id: ObjectID,
     ) -> RpcResult<Vec<SuiObjectInfo>>;
 
+    /// Return the list of dynamic field objects owned by an object.
+    #[method(name = "getDynamicFields")]
+    async fn get_dynamic_fields(
+        &self,
+        /// The ID of the parent object
+        parent_object_id: ObjectID,
+        /// Optional paging cursor
+        cursor: Option<ObjectID>,
+        /// Maximum item returned per page
```

Comment:
nit: looks like otherwise QUERY_MAX_RESULT_LIMIT will be returned, let's add a comment for that?